packages/fxa-shared/speed-trap/performance-factory.js

```diff
@@ -79,10 +78,15 @@ export function getRealPerformanceApi () {
  * of it to support minimal functionality required by speed trap.
  */
 export function getPerformanceApi() {
-  // eslint-disable-next-line no-undef
-  if (!!window.performance && typeof window.performance.now === 'function') {
-    return getRealPerformanceApi();
-  }
+  const api = getRealPerformanceApi();
+  // If the api can produce a time origin and a valid now, let's use it.
+  try {
+    const check = api.timeOrigin + api.now();
```

Comment:
In Sentry, I could observe situations where the flushTime metric was not a valid number. The only way this could happen is if we attempted to use the window.performance api and it wasn't producing valid values, since  `speedtrap.now()` is the result of `performance.timeOrigin + performance.now()`. 

This code attempts to provide a stricter checking by not only making sure `window.performance.now and window.timeOrigin` exist, but also making sure that a valid 'now' value can be determined. If it cannot do so, we will have to fallback to a more generalized api that might be effected hibernates and sleeps.